### PR TITLE
Adds support for visibility in  SearchFilter

### PR DESCRIPTION
**Pull request by jneelamekam**
Updates search filter to include visibility.

### CODE REVIEW

File: src/dispatch/static/dispatch/src/search/SearchFilterCreateDialog.vue [hunk 1]
```diff
@@ -68,6 +68,18 @@
                           <incident-status-multi-select v-model="filters.status" />
                         </v-list-item-content>
                       </v-list-item>
+                      <v-list-item>
+                        <v-list-item-content>
+                          <v-select
+                            :items="visibilities"
+                            v-model="filters.visibility"
```
Comment (mvilanova) [inline src/dispatch/static/dispatch/src/search/SearchFilterCreateDialog.vue:75]: `visibility` is missing in the `filters` dict in the `data()` function and the `this.$watch` at the end.

File: src/dispatch/static/dispatch/src/search/utils.js [hunk 2]
```diff
@@ -97,7 +97,12 @@ export default {
         })
       }
       if (subFilter.length > 0) {
-        filterExpression.push({ or: subFilter })
+       //include "and" condition to filter with visibility
+        if (key == "visibility") {
+          filterExpression.push({ and: subFilter })
+        } else {
+          filterExpression.push({ or: subFilter })
+        }
```
[the remark this answers is not in the document]
Comment (mvilanova) [reply src/dispatch/static/dispatch/src/search/utils.js:105]: I wonder if there's a better way of doing this that does not involve the need of hardcoding a value in a generic function.

File: src/dispatch/plugins/dispatch_slack/dialogs.py [hunk 3]
```diff
@@ -90,9 +90,9 @@ def create_engage_oncall_dialog(
         oncall_service_options.append(
             {"label": oncall_service.name, "value": oncall_service.external_id}
         )
-
+    # sort on-call service options alphabetically
+    oncall_service_options.sort(key=lambda x: x["label"])
```
[the remark this answers is not in the document]
Comment (jneelamekam) [reply src/dispatch/plugins/dispatch_slack/dialogs.py:94]: Will do, I will remove it from this PR and update the code on a separate one.

File: src/dispatch/static/dispatch/src/search/SearchFilterCreateDialog.vue (hunk 1 above)
Comment (jneelamekam) [reply src/dispatch/static/dispatch/src/search/SearchFilterCreateDialog.vue:75]: Done

File: src/dispatch/plugins/dispatch_slack/dialogs.py [hunk 4]
```diff
@@ -92,7 +92,6 @@ def create_engage_oncall_dialog(
         )
 
     page_options = [{"label": "Yes", "value": "Yes"}, {"label": "No", "value": "No"}]
-
```
Comment (mvilanova) [inline src/dispatch/plugins/dispatch_slack/dialogs.py:95]: I think this blank line is fine and doesn't need to be removed.

**Review by mvilanova — APPROVED** [review]

**Pull request merged**